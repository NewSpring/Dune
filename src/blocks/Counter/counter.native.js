//@flow
import { Text, View, Platform, Button } from "react-native";

import type { ICounter } from "./";
import { counterStyle } from "./";

export const Counter = ({
  increment,
  counter,
  decrement,
  loading,
  sample,
}: ICounter) => (
  <View style={counterStyle.resolve()}>
    {loading && <Text>loading...</Text>}
    {sample &&
      <Text>
        GraphQL returned {sample.message} with a response of {sample.code}
      </Text>}
    <Text>Counter is at {counter} on {Platform.OS}</Text>
    <Button onPress={increment} title="Increment" />
    <Button onPress={decrement} title="Decrement" />
  </View>
);
Counter.displayName = "Counter";

export default Counter;
