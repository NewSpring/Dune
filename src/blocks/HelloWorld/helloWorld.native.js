// @flow
import { Text, View } from "react-native";
import React from "react";
import { helloStyle } from "./";
import { Auth } from "../Biometrics";

export const HelloWorld = () => (
  <View>
    <Text style={helloStyle.resolve()}>Hello World</Text>
    <Auth />
  </View>
);
HelloWorld.displayName = "HelloWorld";

export default HelloWorld;
