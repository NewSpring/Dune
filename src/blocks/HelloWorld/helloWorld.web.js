// @flow
import { helloStyle } from "./";

export const HelloWorld = () => (
  <h1 style={helloStyle.resolve()}>Hello World</h1>
);

export default HelloWorld;
