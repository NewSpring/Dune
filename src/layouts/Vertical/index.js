// @flow

import Style from "@jongold/further";
import Vertical from "./vertical";

export const layoutStyle = Style.of({
  flexDirection: "column",
  alignItems: "center",
  height: "50%",
});

export default Vertical;
