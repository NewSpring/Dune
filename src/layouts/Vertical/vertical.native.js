// @flow
import { View, Text } from "react-native";

import { render } from "../";
import { layoutStyle } from "./";

import type { ILayoutProps } from "../../runtime/registry/util/types";

export default ({ zones }: ILayoutProps) => (
  <View>
    <View>
      <Text>
        Native Vertical Layout
      </Text>
    </View>
    <View style={layoutStyle.resolve()}>

      {/* main zone */}
      <View>
        {zones.main && zones.main.map(render)}
      </View>

      {/* secondary zone */}
      <View>
        {zones.secondary && zones.secondary.map(render)}
      </View>

    </View>
  </View>
);
