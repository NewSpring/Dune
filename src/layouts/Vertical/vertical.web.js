// @flow

import { render } from "../";
import { layoutStyle } from "./";

import type { ILayoutProps } from "../../runtime/registry/util/types";

export default ({ zones }: ILayoutProps) => (
  <div>
    <div>
      Web Vertical Layout
    </div>
    <div style={layoutStyle.resolve()}>
      <div>
        {zones.main && zones.main.map(render)}
      </div>
      {/* secondary zone */}
      <div>
        {zones.secondary && zones.secondary.map(render)}
      </div>
    </div>
  </div>
);
