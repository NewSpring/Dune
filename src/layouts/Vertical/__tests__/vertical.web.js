import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Vertical from "../vertical.web.js";
import Counter from "../../../blocks/Counter";
import HelloWorld from "../../../blocks/HelloWorld";

const layoutProps = {
  zones: {
    main: [
      { id: 1, path: "Counter", Component: Counter },
      { id: 2, path: "HelloWorld", Component: HelloWorld },
    ],
    secondary: [{ id: 2, path: "HelloWorld", Component: HelloWorld }],
  },
};

const generateComponent = additionalProps => (
  <Vertical {...layoutProps} {...additionalProps} />
);

describe("Vertical Web Layout", () => {
  it("should render Vertical layout component", () => {
    const tree = shallow(generateComponent());
    expect(tree).toBeDefined();
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
