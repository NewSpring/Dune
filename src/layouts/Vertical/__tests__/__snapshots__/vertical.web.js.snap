// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Web Layout should render Vertical layout component 1`] = `
<div>
  <div>
    Web Vertical Layout
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "50%",
      }
    }
  >
    <div>
      <Apollo(withState(withHandlers(Counter)))
        Component={[Function]}
        id={1}
        path="Counter"
      />
      <HelloWorld
        Component={[Function]}
        id={2}
        path="HelloWorld"
      />
    </div>
    <div>
      <HelloWorld
        Component={[Function]}
        id={2}
        path="HelloWorld"
      />
    </div>
  </div>
</div>
`;
